crates/ruma-client-api/src/error.rs

```diff
@@ -249,7 +253,24 @@ impl EndpointError for Error {
     ) -> Result<Self, DeserializationError> {
         let status = response.status();
         let error_body: ErrorBody = from_json_slice(response.body().as_ref())?;
-        Ok(error_body.into_error(status))
+
+        #[cfg(not(feature = "unstable-msc2967"))]
+        {
+            Ok(error_body.into_error(status))
+        }
+
+        #[cfg(feature = "unstable-msc2967")]
+        {
+            let mut error = error_body.into_error(status);
+
+            error.authenticate = response
+                .headers()
+                .get(http::header::WWW_AUTHENTICATE)
+                .and_then(|val| val.to_str().ok())
```

Comment:
I think `.to_str()` failing should be a `DeserializationError`, rather than a successfully-deserialized `Error` with `authenticate: None`.